src/app/shared/iframe/iframe.ts

```diff
@@ -0,0 +1,10 @@
+export enum BubbleToIonicPostMessage {
+  IFRAME_ON_LOAD = 'iframe-on-load',
+  IFRAME_BACK_BUTTON_CLICKED = 'iframe-back-button-clicked',
+  EDIT_CAPTION_CANCEL = 'edit-caption-cancel',
+  EDIT_CAPTION_SAVE = 'edit-caption-save',
+}
+
+export enum IonicToBubblePostMessage {
+  ANDROID_BACK_BUTTON = 'android-back-button',
+}
```

Comment:
**by ChatGPT**
The code seems to be a refactoring of the way post message events are handled in iframes, with the introduction of enums and switch statements. Here are some suggestions:

-   It is good practice to have clear and concise commit messages. Instead of "refactor: to enums + switch for iframe post message events," consider something like "Refactor iframe post message event handling with enums and switch statements."
-   In the `BubbleToIonicPostMessage` enum, the values should be in uppercase letters, separated by underscores, to follow common naming conventions for enums. For example, `IFRAME_ON_LOAD` instead of `iframe-on-load`.
-   Consider adding comments to explain the purpose of the enums and the values they contain, especially if they are used across multiple files and/or projects.
-   In the `EditCaptionPage` and `TermsOfUsePage`, the `postMessageEvent.data` is being cast to the `BubbleToIonicPostMessage` type without any error handling in case the value is not a member of the enum. It may be a good idea to add a default case to the switch statement that logs an error message and/or throws an error if the value is not recognized.
-   Consider using a more descriptive name for the enums, especially if they are used in multiple contexts. For example, instead of `BubbleToIonicPostMessage`, consider something like `IframeEvent`.

Reply:
@tammyyang :
- opinionated
- opinionated although good suggestion but it will require changes from bubble side as well so would keep lower case + underscore
- asana [card](https://app.asana.com/0/1201016280880500/1204028658078827/f) created.
- opinionated